pkg/controller/operators/catalog/operator.go

```diff
@@ -1347,11 +1347,22 @@ type UnpackedBundleReference struct {
 	Properties             string `json:"properties"`
 }
 
-// unpackBundles makes one walk through the bundlelookups and attempts to progress them
-func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.InstallPlan, error) {
+/* unpackBundles makes one walk through the bundlelookups and attempts to progress them
+Returns:
+    unpacked: bool - If the bundle was successfully unpacked
+    out: *v1alpha1.InstallPlan - the resulting installPlan
+    errorWasFatal: bool - If a returned error is fatal and should result in IP failure
+	error: error
+*/
+
+func (o *Operator) unpackBundles(plan *v1alpha1.InstallPlan) (bool, *v1alpha1.InstallPlan, bool, error) {
 	out := plan.DeepCopy()
 	unpacked := true
 
+	// We should return whether the error which caused failure was transient (install should retry) or
+	// fatal (it should not)
+	errorWasFatal := false
```

Comment:
`isFatal` would be sufficient for variable name. Also `var isFatal bool` would be good as well.

Reply:
var isFatal bool definitely does a better job of communicating what is going on in a readable way. Will implement.

Reply:
To begin I think I may need guidance on which errors should be considered "retryable" and which should not.

Reply:
I have implemented a basic version of this. Let me know how it looks.